naptime/src/main/scala/org/coursera/naptime/Types.scala

```diff
@@ -115,11 +115,15 @@ object Types extends StrictLogging {
     val errorMessageBuilder = new StringBuilder
     val recordDataSchema = new RecordDataSchema(new Name(typeName), RecordType.RECORD)
     val fieldsMinusId = keyType.getFields().asScala.toList ++ valueType.getFields().asScala
-    val idField = new RecordDataSchema.Field(new StringDataSchema)
-    idField.setOptional(false)
-    idField.setName("id", errorMessageBuilder)
-    idField.setRecord(recordDataSchema)
-    val fields = idField :: fieldsMinusId
+    val fields = if (fieldsMinusId.exists(_.getName == "id")) {
+      fieldsMinusId // The key type had an `id` field defined as part of it.
```

Comment:
i'm not crazy about the name `fieldsMinusId`, since its not really fields minus id. Can we just call this "combinedFields" instead maybe?